logstash-output-opensearch.gemspec

```diff
@@ -18,7 +20,10 @@ Gem::Specification.new do |s|
   # Tests
   s.test_files = s.files.grep(%r{^(test|spec|features)/})
 
-  s.cert_chain  = ['certs/opensearch-rubygems.pem']
+  if $PROGRAM_NAME.end_with?("gem") && ARGV == ["build", __FILE__] && File.exist?(signing_key_path)
```

Comment:
@naveentatikonda Could you explain what this is doing? Im not very familiar with gem signing.

Reply:
To avoid security issues while installing gems where if we install any malicious gem without knowing its credibility, the gem author can penetrate into that server. To avoid this, the RubyGems has an ability to sign the gems where the signing data is included as part of the gem. So that the end user can verify the signing key before installing the gem.

Reply:
I am still confused by this is going from being not conditional to conditional on who's calling.

Reply:
Certificate  is useful only if we are signing the gems. Hence, we moved this attribute to be added only if private key exists